File: create_and_delete_directory/create_and_delete_directory.js
let fs = require("fs");
let fsExtra = require('fs-extra');
const root_dir = './example_directoy';
const lang_dir = 'js';
const dirpath = [root_dir, lang_dir].join("/");

fs.mkdir(dirpath, { recursive: true }, (err) => {
  if (err) {
    throw err;
  }
  else{
    console.log("Succeeded in creation of " + root_dir);
    if(root_dir.match(/[a-zA-Z\d_\-]/)) {
      fsExtra.remove(root_dir, (err) => {
        if (err) {
          throw err;
        }
        else{
          console.log("Succeeded in removal of " + root_dir);
        }
      });
    }
  }
});

File: examples/file/create_and_delete_directory/create_and_delete_directory.pl
use File::Path qw(make_path remove_tree);

my $root_dir = './example_directoy';
my $lang_dir = 'perl';
my $dirpath = join('/', $root_dir, $lang_dir);
unless(-d $dirpath){
  make_path($dirpath);
}
if(-d $dirpath) {
  print "Succeeded in creation of ".$dirpath."\n";
  if($root_dir=~ m![a-zA-Z\d_\-]!){
    remove_tree($root_dir);
  }
  if(-d $root_dir) {
    die("Failed to remove ".$root_dir)
  }
  else{
    print "Succeeded in removal of ".$dirpath."\n";
  }
}
else{
  print "Failed to mkdir -p ".$dirpath."\n";
}